### PR TITLE
Prevent multiple versions of Vue/Pinia

### LINKED ISSUE #20064: Theming error in production
### Describe the Bug

When setting up Directus with the `next` branch for the first time (aka new database, no extensions and running in production mode). We are getting errors in Theming, what looks like an incorrect font and the same Dark mode issues mentioned in https://github.com/directus/directus/issues/20044

This seems to only happen in production so getting to the source of those console errors will take some debugging.

Darkmode Firefox:
firefox_m94vIGAIEd
image

Lightmode Chrome:
image
image

In development mode everything looks fine (using the same API/DB):
image
image

### To Reproduce

1. Create a new instance with a new database
2. Run the instance in production mode
3. Navigate to the login screen in Firefox or Chrome
4. [optional] Login

### Directus Version

next

### Hosting Strategy

Self-Hosted (Custom)

### PR DESCRIPTION
**Pull request by paescuj**
## Scope

What's changed:

- Align Pinia version within the codebase
- Declare Vue/Pinia dependencies as peer dependencies in sub packages to prevent multiple instances/versions being used

## Potential Risks / Drawbacks

none

## Review Notes / Questions

- IMO the peer dependency in `@directus/utils` can be optional, as the package also exports node-only utils

---

Fixes #20064

### CODE REVIEW

**Comment by br41nslug** [comment]
Wow of course it had to be something like this 😂

File: packages/components/package.json [hunk 1]
```diff
@@ -43,7 +42,7 @@
 		"vite-plugin-dts": "2.3.0",
 		"vitest": "0.31.1"
 	},
-	"dependencies": {
+	"peerDependencies": {
 		"pinia": "2.1.1",
 		"vue": "3.3.4",
 		"vue-i18n": "9.2.2",
```
Comment (br41nslug) [inline packages/components/package.json:48]: Not sure if we can move `vue` and `vue-i18n` to peerDependencies 🤔 I thought there was a reason for at least `vue` needing to be in `dependencies`.

File: packages/stores/package.json [hunk 2]
```diff
@@ -24,15 +24,17 @@
...
 		"tsup": "7.2.0",
 		"typescript": "5.2.2"
 	},
-	"dependencies": {
-		"@vueuse/core": "10.1.2",
-		"pinia": "2.1.1",
+	"peerDependencies": {
+		"pinia": "2.1.7",
 		"vue": "3.3.4"
 	}
```
Comment (br41nslug) [inline packages/stores/package.json:39]: Same hesitation about `vue` as above.

**Review by br41nslug — COMMENTED** [review]
I can confirm that this fixes the linked `pinia` issue
image

File: packages/components/package.json (hunk 1 above)
Comment (br41nslug) [reply packages/components/package.json:48]: might be outdated knowledge on my part tho 🤔

Comment (paescuj) [reply packages/components/package.json:48]: I cannot think of any 🤔🤔

Comment (paescuj) [reply packages/components/package.json:48]: (cc @nickrum 😇)

**Review by br41nslug — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
Great find @paescuj! 👏🏻👏🏻

File: packages/components/package.json (hunk 1 above)
Comment (nickrum) [reply packages/components/package.json:48]: I think it depends on the package manager you install Directus with and whether the package `@directus/components` depends on has `vue` and `vue-i18n` listed as dependencies. As long as those dependencies can be resolved by Node, it should be fine using any package manager. If not, it depends on whether the package manager installs peer dependencies by default if they are not present.

By the way, we had issues with different `vue` versions being used at the same time before. That's why `vue` and a couple of others were peer dependencies of `@directus/types` with no exact version specified for quite some time.

**Pull request merged**